### ISSUE TITLE
Tests required for tables with $table_name declared

### ISSUE THREAD

**Issue by zanderwar** [UndefinedOffset/SortableGridField#105]
I'm currently migrating this module over to SS4 and have discovered:

MySQL has a max table name of 64 characters and when appending the FQCN to the table as required by SS4, these tables will not be built.

Therefore it is required to add `private static $table_name` to remove the FQCN from the table name and as the `UPDATE` calls don't check for the existence of a `table_name` configuration option, the tests failed, heres my fix:

```
$table = (Config::inst()->get($table, 'table_name', CONFIG::UNINHERITED)) ? Config::inst()->get($table, 'table_name', CONFIG::UNINHERITED) : $table;
```

After adding this _everywhere_ it's required, this fixes the issue with `DataObjects` that have declared `$table_name`

~The reason I'm bringing this up is it because it also looks like this bug would exist in the SS3 version~ [scrap that] $table_name isn't introduced until version 4

**Comment by zanderwar** [comment]
Would you like a seperate issue for the bug? The update queries won't occur in SS3 if `DataObject`s have `$table_name` declared

**Comment by UndefinedOffset** [comment]
Sorry miss tagged this I think this does sound like a bug to me after reading closer

**Comment by zanderwar** [comment]
This might require a map:

```
$array = [
      'GridFieldAction_SortOrder_Team' => GridFieldAction_SortOrder_Team::class
];
```

As you do some `class_exists($table)` when dealing with `Versioned` and if `$table` has become what is declared in `$table_name` then this `Versioned` functionality will never fire.

**Comment by zanderwar** [comment]
This is not even available until SS4, closing issue as resolved in #106
